pages/shoppingcart.js

```diff
@@ -1,27 +1,29 @@
-function ShoppingCart({ ingredients, onUpdateIngredients }) {
-  const selection = ingredients.filter((ingredient) => ingredient.chosen);
+import useLocalStorage from '../hooks/useLocalStorage';
+
+function ShoppingCart() {
+  const [selectedIngredients, setSelectedIngredients] = useLocalStorage(
+    '_ingredients',
+    []
+  );
 
   function removeBagle(id) {
-    const updatedIngredients = ingredients.map((ingredient) => {
-      if (ingredient.id === id) {
-        ingredient.chosen = false;
-      }
-      return ingredient;
-    });
-    onUpdateIngredients(updatedIngredients);
+    const newIngredients = selectedIngredients.filter(
+      (ingredient) => ingredient.id !== id
+    );
+    setSelectedIngredients(newIngredients);
   }
 
   return (
     <>
       <h2>Shopping Cart</h2>
 
       <ul>
-        {selection.map((ingredient) => {
+        {selectedIngredients.map((ingredient) => {
           return (
-            <>
-              <li key={ingredient.id}>{ingredient.name}</li>;
+            <div key={ingredient.id}>
```

Comment:
Instead of working with a `div` you could also use Fragments like you can see in the [React docs](https://reactjs.org/docs/fragments.html#keyed-fragments).